lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,28 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.read(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            s = input.readpartial(4096)
```

Comment:
`buf.rpartition(/[\r\n]/)` more likely.

and I'd need an inner loop, because we could've read in more than one `\n` in that particular `readpartial`, no?